scripts/run_backend_tests.py

```diff
@@ -404,29 +299,28 @@ def main(args=None):
             test_path=parsed_args.test_path,
             include_load_tests=include_load_tests)
 
-    # Prepare tasks.
-    concurrent_count = min(multiprocessing.cpu_count(), MAX_CONCURRENT_RUNS)
-    semaphore = threading.Semaphore(concurrent_count)
+
 
     task_to_taskspec = {}
     tasks = []
     for test_target in all_test_targets:
         test = TestingTaskSpec(
             test_target, parsed_args.generate_coverage_report)
-        task = TaskThread(
-            test.run, parsed_args.verbose, semaphore, name=test_target)
+        task = semaphore_utils.TaskThread(
```

Comment:
Perhaps, conceptually, this should be semaphore_utils.create_task(...)? That way, you have a nice clean reusable API, with create_task(), execute_tasks(), etc.

Reply:
Done.